### PR TITLE
Add signals and hooks to allow multiple affiliations

### PR DESCRIPTION
**Pull request by duartegalvao**
This PR adds the necessary signals and hooks to allow the implementation of multiple affiliations per contribution author, implemented in indico/indico-plugin-jacow#8.

### CODE REVIEW

File: indico/core/signals/event/abstracts.py [hunk 1]
```diff
@@ -23,3 +23,9 @@
 abstract_updated = _signals.signal('abstract-updated', '''
```
Comment (duartegalvao) [inline indico/core/signals/event/abstracts.py:23]: This signal is not called anywhere - should I delete it?

Comment (tomasr8) [reply indico/core/signals/event/abstracts.py:23]: Or call it in `update_abstract`? :thinking:

Comment (duartegalvao) [reply indico/core/signals/event/abstracts.py:23]: I can, but is it actually needed for anything? 😅

Comment (tomasr8) [reply indico/core/signals/event/abstracts.py:23]: dunno :sweat_smile:

Comment (ThiefMaster) [reply indico/core/signals/event/abstracts.py:23]: seems useful to have it, even if nothing is using it right now

File: indico/web/client/js/react/components/PersonLinkField.jsx [hunk 2]
```diff
@@ -281,6 +287,7 @@ function PersonLinkField({
     values.forEach(p => {
       p.name = formatName(p);
       p.roles = roles.filter(x => x.default).map(x => x.name);
+      getPluginObjects('onAddPersonLink').forEach(f => f(p));
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/PersonLinkField.jsx:290]: I'd do `const hooks =  getPluginObjects('onAddPersonLink');` outside the `values.forEach()` loop and then just run `hooks.forEach(...)` here. No need to build that list every time...

**Review by ThiefMaster — APPROVED** [review]

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**